controllers/webhook/storageclass.go

```diff
@@ -38,7 +43,86 @@ func (s *StorageClassAdmission) Handle(ctx context.Context, req admission.Reques
 		return admission.Errored(http.StatusBadRequest, fmt.Errorf("supported provisioners are %s", supportedProvisioners))
 	}
 
-	// TODO: mutate the storageclass with default parameters based on info from storagecluster
+	storageConsumerList := &v1alpha1.StorageConsumerList{}
+	if err := s.List(ctx, storageConsumerList, client.InNamespace(s.Namespace)); err != nil {
+		s.Log.Error(err, "failed to list storageconsumers", "namespace", s.Namespace)
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed to list storageconsumers in %s namespace", s.Namespace))
+	}
+
+	if len(storageConsumerList.Items) == 0 {
+		s.Log.Error(fmt.Errorf("no storageconsumers found in %s namespace", s.Namespace), "failed validation")
+		return admission.Denied("creation of storageclass should happen after storageconsumer exists")
+	}
+
+	clusterID := ""
+	for idx := range storageConsumerList.Items {
+		consumer := &storageConsumerList.Items[idx]
+		if consumer.Annotations[defaults.StorageConsumerTypeAnnotation] == defaults.StorageConsumerTypeLocal {
+			clusterID = string(consumer.UID)
+			break
+		}
+	}
+
+	if clusterID == "" {
+		s.Log.Error(fmt.Errorf("no storageconsumer is marked as local in %s namespace", s.Namespace), "failed validation")
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed to get local storageconsumer in %s namespace", s.Namespace))
+	}
+
+	s.Log.Info("populating default parameters", "storageclass", storageClass.Name, "provisioner", storageClass.Provisioner)
+	var provisionerSecretName, nodeStageSecretName string
+	if storageClass.Provisioner == util.RbdDriverName {
+		provisionerSecretName = getSecretName("rbd", "provisioner", clusterID)
+		nodeStageSecretName = getSecretName("rbd", "node", clusterID)
+	} else if storageClass.Provisioner == util.CephFSDriverName {
+		provisionerSecretName = getSecretName("cephfs", "provisioner", clusterID)
+		nodeStageSecretName = getSecretName("cephfs", "node", clusterID)
+	}
+
+	patches := []jsonpatch.JsonPatchOperation{}
+	if len(storageClass.Parameters) != 0 {
+		patches = append(patches, jsonpatch.JsonPatchOperation{
+			Operation: "add",
+			Path:      "/parameters",
+			Value:     map[string]string{},
+		})
+		s.Log.Info("adding storageclass parameters section")
+	}
+	if storageClass.Parameters["csi.storage.k8s.io/provisioner-secret-name"] != "" {
+		patches = append(patches, jsonpatch.JsonPatchOperation{
+			Operation: "add",
+			Path:      "/parameters/csi.storage.k8s.io~1provisioner-secret-name",
```

Comment:
`csi.storage.k8s.io~1provisioner-secret-name` typo in this one

Reply:
i felt it as typo, can you please add a comment for this one :(

Reply:
done.